tests/lib/prepare-project.sh

```diff
@@ -66,6 +66,42 @@ fedora_build_rpm() {
     rm $GOPATH/*.src.rpm
 }
 
+opensuse_build_rpm() {
+    release=$(echo "$SPREAD_SYSTEM" | awk '{split($0,a,"-");print a[2]}')
+    arch=x86_64
+
+    base_version="$(head -1 debian/changelog | awk -F'[()]' '{print $2}')"
+    version="1337.$base_version"
+    sed -i -e "s/^Version:.*$/Version: $version/g" packaging/opensuse-$release/snapd.spec
+
+    mkdir -p /tmp/pkg/snapd-$version
+    cp -rav * /tmp/pkg/snapd-$version/
```

Comment:
A `osc build` isn't possible without a login. Also I have code to unite the rpm build for Fedora and openSUSE under a single `build_rpm` function.